README.md

```diff
@@ -1 +1,571 @@
+
+#### List Fields
+
+Fields that return lists or collection wrappers (like Relay connections) should be plural.
+
+```graphql
+# good
+type User {
+  memberships: [Membership!]
+}
+
+# bad
+type User {
+  membership: [Membership!]
+}
+```
+
+### Documentation
+
+Use GraphQL descriptions everywhere: types, fields, arguments, etc.
+
+Do not make front-end developers guess the meaning or functionality of the schema; tell them.
+
+### Custom Scalars
+
+Use custom scalars (in moderation).
+
+Custom scalars allow schemas to be extended with new scalars that are outside of the GraphQL spec. Use this feature, but only for well-understood domains that offer significant leverage.
+
+Good examples:
+- `ISO8601DateTime`
+- `E164PhoneNumber`
+- `EmailAddress`
+- `URL`
+- `JSON`
+
+Bad examples:
+- `Name` - just use String
+- `Char32` - just use String
+
+### Field Names and Types
+
+Prefer very specific field names or nested object types.
+
+Use very descriptive and specific names, especially while the domain is not yet well-understood. This allows the use of more general names when the domain is more understood. Typically, general names are best reserved for interfaces and unions.
+
+Alternatively, if specific names do not work or are unwieldy, nest types instead. For example, the following two schemas are roughly the same in terms of flexibility.
+
+```graphql
+type User {
+  roleName: String
+}
+```
+
+```graphql
+type User {
+  role: UserRole
+}
+
+type UserRole {
+  name: String
+}
+```
+
+This extends to scalars as well. An email could be represented as a string, or as an object, like this:
+
+```graphql
+type Email {
+  address: EmailAddress!
+  isVerified: Boolean!
+  verifiedAt: ISO8601DateTime
+}
+```
+### Interfaces and Unions
+
+Unions are semantically just marker interfaces (interfaces with zero fields) so I will refer to both interfaces and unions as interfaces.
+
+When choosing field types, prefer interfaces instead of concrete types.
+
+Instead of a `Comment` type, consider a `Comment` interface and a `PostComment` type. Instead of a `Message` type, consider a `Message` interface and a `ChatMessage` type.
+
+### References
+
+Refer to objects, not IDs. GraphQL allows direct traversal so let clients do that.
+
+```graphql
+# good
+type User {
+  invitedBy: User
+}
+
+# bad
+type User {
+  invitedBy: ID
+}
+```
+
+### Relay
+
+Implement the Relay spec. Read [this](https://relay.dev/docs/en/graphql-server-specification).
+
+One thing often neglected in Relay's cursor pagination is that there is nothing preventing the extension of connections, edges, and page info with additional fields. Cursor pagination can get unwieldy with nested associations. However, this can be avoided by treating your data model as nodes and edges, instead of only using edges for pagination. For example, organization memberships are conceptually edges between users and organizations and should be able to be queried as such.
+
```

Comment:
Before going into a discussion about what to do with Relay, maybe give a quick one or two sentence blurb about what Relay is.

Ex:
```
Relay couples React with GraphQL. It allows React components to specify what data they need and the Relay framework provides that data.

For implementing the Relay spec, read...
```